### ISSUE TITLE
Add feature flags for network protocol level changes.

### ISSUE THREAD

**Issue by iansuvak** [ava-labs/awm-relayer#440]
**Context and scope**
Current features being developed such as ACP-118 support #383 depend on network level changes that are present in default branches in upstream repos (`avalanchego`, `coreth` and `subnet-evm`).

We need to be able to continue developing on top of these new features while also preserving the ability to push hot fixes and new versions to currently deployed instances of relayer. 

To do so we need a configuration setting that determines which protocol level of the network is required for given features. 

**Discussion and alternatives**
Alternatives that were proposed were:
- Maintained a long-lived dev branch that wouldn't get merged into main until the network upgrade
- Maintain separate hotfix release branches that would get updated with only the code that needs to get updated for a new feature or bugfix before the next network update. 

Both of these seem more cumbersome and error prone than a re-usable feature flag config settings that defaults to the most recent network version live on mainnet.

**Comment by iansuvak** [comment]
Implemented in #463 -- Can close once #408 is merged
